Jenkinsfile

```diff
@@ -286,19 +287,35 @@ try {
           trendChartType: 'NONE'
         )
       }
-    }
 
-    timeout(time: 10, unit: 'MINUTES') {
-      def qualityGate = waitForQualityGate()
-      if (qualityGate.status != 'OK') {
-        currentBuild.result = 'FAIL'
+    def reportFilePath = "target/sonar/report-task.txt"
+    def reportTaskFileExists = fileExists "${reportFilePath}"
+    if (reportTaskFileExists) {
+      echo "Found report task file"
+      def taskProps = readProperties file: "${reportFilePath}"
+      echo "taskId[${taskProps['ceTaskId']}]"
+      timeout(time: 10, unit: 'MINUTES') {
+        while (true) {
+          sleep 10
+          def taskStatusResult    =
+          sh(returnStdout: true, script: "curl -s -X GET -u ${authString} \'${sonarProps['sonar.host.url']}/api/ce/task?id=${taskProps['ceTaskId']}\'")
+          echo "taskStatusResult[${taskStatusResult}]"
+          def taskStatus  = new JsonSlurper().parseText(taskStatusResult).task.status
+          echo "taskStatus[${taskStatus}]"
+          // Status can be SUCCESS, ERROR, PENDING, or IN_PROGRESS. The last two indicate it's
+          // not done yet.
+          if (taskStatus != "IN_PROGRESS" && taskStatus != "PENDING") {
+            break;
+          }
+          def qualityGate = waitForQualityGate()
+          if (qualityGate.status != 'OK') {
+            currentBuild.result = 'FAIL'
+          }
+        }
       }
     }
-
-    if ((currentBuild.result ?: 'SUCCESS') != 'SUCCESS') {
-      error("Quality gate failure: ${qualityGate.status}.");
-    }
```

Comment:
Isn't this check required to impact the lines 267, 273 or 282 results on the build ?

Reply:
True. Reintroduced in the next commit 7a05b8d.